src/window_message.cpp

```diff
@@ -105,13 +106,20 @@ void Window_Message::ApplyTextInsertingCommands() {
 				bool success;
 				int parsed_num;
 				std::u32string command_result = Utils::DecodeUTF32(ParseCommandCode(success, parsed_num));
+				if (start_code < actor_replacement_start) {
+					replaced_actors.clear();
+				}
+
 				if (!success || std::find(replaced_actors.begin(), replaced_actors.end(), parsed_num) != replaced_actors.end()) {
 					text_index = start_code - 2;
 					continue;
 				}
 
 				if (ch == 'n') {
 					replaced_actors.push_back(parsed_num);
+					if (actor_replacement_start >= text_index - 1) {
```

Comment:
I think the invalid iterator is really the problem.
Changing actor_replace_start = end; to =start;

and the code to:
```cpp
				long actor_replacement_dif = 0;
				if (ch == 'n') {
					replaced_actors.push_back(parsed_num);
					if (actor_replacement_start <= text_index) {
						actor_replacement_dif = std::distance(text.begin(), text_index);
					}
				}

				text.replace(start_code, text_index + 1, command_result);
				// Start from the beginning, the inserted text might add new commands
				text_index = text.end();
				end = text.end();

				actor_replacement_start = text.begin();
				actor_replacement_start += actor_replacement_dif;
```
Basicly remembering the iterator distance in the old code and moving the iterator this distance in the new.

fixes the problem I mentioned for me. Could you retest?

But the following is broken now: When an actor is called \N[5] and 4 is called \N[4] :/